library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(ggplot2)
library(biomaRt)
require(plyr)
library(magrittr)

setwd("X:/0-PRJ/1111-CTRP6/Analysis/04.WT2KO/01.MFI/01.deg")

############# functions ###############
# dds: dds file generated by DESeq(dds)
# x: the nominator
# y: the demoninator (ie. the reference)
get_de_lst <- function(dds, x, y, prefix){
  res <- results(dds, contrast = c("Group", x, y))
  de <- res[!is.na(res$log2FoldChange) & !is.na(res$padj) & res$padj < 0.05,]
  de <- de[order(de$pvalue),]
  up <- de[de$log2FoldChange > 1,]
  dn <- de[de$log2FoldChange < -1,]
  write.table(
    res, file = paste0('Table.',prefix,'.al.inf.txt'),
    sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE
  )
  write.table(
    up, file = paste0('Table.',prefix,'.up.inf.txt'),
    sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE
  )
  write.table(
    dn, file = paste0('Table.',prefix,'.dn.inf.txt'),
    sep = "\t", quote = FALSE, row.names = TRUE, col.names = TRUE
  )
  write.table(
    sub(":.*$","",row.names(up)), file = paste0('Table.',prefix,'.up.ID.txt'),
    sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE
  )
  write.table(
    sub(":.*$","",row.names(dn)), file = paste0('Table.',prefix,'.dn.ID.txt'),
    sep = "\t", quote = FALSE, row.names = FALSE, col.names = FALSE
  )
  lst <-  list(
    al = res,
    up = up,
    dn = dn,
    up.lst = row.names(up),
    dn.lst = row.names(dn),
    up.num = nrow(up),
    dn.num = nrow(dn)
  )
  
  plot_MA(lst, prefix)
  
  return(lst)
}

## plot_MA
plot_MA <- function(lst, name){
  svg(file = paste0("Fig.MA_plot.", name, ".svg"), width = 4, height = 3.6)
  par(mar=c(4,4,1,1))
  plot(
    log10(lst$al$baseMean+1), lst$al$log2FoldChange, 
    cex = .1, pch = 20, col = "grey", xaxt = "n", xlim = c(0,6), ylim = c(-8,8),
    xlab = "Averaged expression", ylab = "log2(KO/WT)", main = name
  )
  cl <- brewer.pal(8, "Dark2")[2]
  axis(side = 1, at = 0:6, labels = c(1, 10, 100, "1e3", "1e4", "1e5", "1e6"))
  abline(h=0, lwd = 3, col = "orange")
  points(
    log10(lst$al$baseMean[row.names(lst$al) %in% c(lst$up.lst,lst$dn.lst)]+1), 
    lst$al$log2FoldChange[row.names(lst$al) %in% c(lst$up.lst,lst$dn.lst)],
    col = cl, pch = 19, cex = 0.5
  )
  legend("topright",    bty = "n", legend = paste0("Increased: ", lst$up.num))
  legend("bottomright", bty = "n", legend = paste0("Decreased: ", lst$dn.num))
  dev.off()
}

d <- read.table("merged_count.txt", header=T, row.names = 1)[,-1:-5]
ids <- row.names(d)
row.names(d) <- paste0(row.names(d), ":", d$gene_name)
d <- d[,-1]

## rename colnames
colnames(d) <- sub(".bam","",colnames(d)) %>% sub("MFI_P", "gd", .) %>% sub("5_", ".5_", .)
d <- d[,c(4:6,1:3,10:12,7:9,16:18,13:15,22:24,19:21)]

## annotation and filter genes
mart <- useMart("ensembl", host="www.ensembl.org")
inf <- getBM(
   attributes=c("ensembl_gene_id","external_gene_name", "chromosome_name", "gene_biotype"),  
   mart= useDataset("mmusculus_gene_ensembl",mart),
   filters="ensembl_gene_id", values=ids
 )
saveRDS(inf, "inf.rds")

inf <- readRDS("inf.rds")
inf.flt <- inf[inf$chromosome_name != "MT" & inf$chromosome_name != "Y",]

d.flt <- d[sub(":.*$","",row.names(d)) %in% inf.flt$ensembl_gene_id,]

## make DESeq dataset
cnt.data <- d.flt[apply(d.flt,1,max)>=10,]
# genes before and after filtering based on counts
nrow(d.flt)
nrow(cnt.data)

groupName <- factor(
  rep(
    c(
      "gd10.5_WT", "gd10.5_KO", "gd14.5_WT", "gd14.5_KO",
      "gd16.5_WT", "gd16.5_KO", "gd19.5_WT", "gd19.5_KO"
      ),
    each = 3
    ),
  levels = c(
    "gd10.5_KO", "gd10.5_WT", "gd14.5_KO", "gd14.5_WT",
    "gd16.5_KO", "gd16.5_WT", "gd19.5_KO", "gd19.5_WT"
    )
)

sampleName <- factor(
  c(
    "gd10.5_WT_1", "gd10.5_WT_2", "gd10.5_WT_3", "gd10.5_KO_1", "gd10.5_KO_2", "gd10.5_KO_3", 
    "gd14.5_WT_1", "gd14.5_WT_2", "gd14.5_WT_3", "gd14.5_KO_1", "gd14.5_KO_2", "gd14.5_KO_3",
    "gd16.5_WT_1", "gd16.5_WT_2", "gd16.5_WT_3", "gd16.5_KO_1", "gd16.5_KO_2", "gd16.5_KO_3",
    "gd19.5_WT_1", "gd19.5_WT_2", "gd19.5_WT_3", "gd19.5_KO_1", "gd19.5_KO_2", "gd19.5_KO_3"
  ),
  levels = c(
    "gd10.5_KO_1", "gd10.5_KO_2", "gd10.5_KO_3", "gd10.5_WT_1", "gd10.5_WT_2", "gd10.5_WT_3", 
    "gd14.5_KO_1", "gd14.5_KO_2", "gd14.5_KO_3", "gd14.5_WT_1", "gd14.5_WT_2", "gd14.5_WT_3",
    "gd16.5_KO_1", "gd16.5_KO_2", "gd16.5_KO_3", "gd16.5_WT_1", "gd16.5_WT_2", "gd16.5_WT_3",
    "gd19.5_KO_1", "gd19.5_KO_2", "gd19.5_KO_3", "gd19.5_WT_1", "gd19.5_WT_2", "gd19.5_WT_3"
    )
)

col.data <- data.frame(
  Group = groupName,
  Sample = sampleName,
  Genotype = sub("^.*_", "", groupName),
  Stage = sub("_.*$", "", groupName)
)

dds <- DESeqDataSetFromMatrix(
  countData = cnt.data, colData = col.data, design = ~ Group
)

## Get normalized counts
dds <- estimateSizeFactors(dds)
dds.cnt.nrm <- counts(dds, normalized = TRUE)
colnames(dds.cnt.nrm) <- dds@colData$Sample
write.table(dds.cnt.nrm, file="Merged_cnt.nrm.xls", sep="\t", quote=FALSE, row.names = TRUE, col.names = TRUE)

#heatmap
svg(file="Fig.heatmap_by_cor.log10.svg", width = 9, height = 8)
pheatmap(
  cor(log10(dds.cnt.nrm+1), method = "p"), 
  display_numbers = TRUE
)
dev.off()


########## PCA analysis ##########
rld <- rlog(dds, blind = TRUE)

## All samples
pca.data <- plotPCA(
  rld, intgroup = c("Stage", "Genotype"),returnData = TRUE, ntop = 1000
)
percent.var <- round(100*attr(pca.data, "percentVar"))

svg(file = "Fig.pca_plot.svg", width = 4.5, height = 2.6)
ggplot(pca.data, aes(PC1, PC2, color=Stage, shape = Genotype)) +
  theme_bw() + 
  geom_point(size = 4, alpha = 0.4, position = "jitter") +
  xlab(paste0("PC1: ", percent.var[1], "% variance")) +
  ylab(paste0("PC2: ", percent.var[2], "% variance")) +
  scale_color_manual(values = brewer.pal(4,"Set1"))
dev.off()

################# Part I: Untreated samples & other basal analyses
dds <- DESeq(dds)

saveRDS(dds, file = "dds.rds")

dds <- readRDS("dds.rds")

## DEGs
de.e10 <- get_de_lst(dds, "gd10.5_KO", "gd10.5_WT", "gd10.5")
de.e14 <- get_de_lst(dds, "gd14.5_KO", "gd14.5_WT", "gd14.5")
de.e16 <- get_de_lst(dds, "gd16.5_KO", "gd16.5_WT", "gd16.5")
de.e19 <- get_de_lst(dds, "gd19.5_KO", "gd19.5_WT", "gd19.5")

## averaged cnt
cnt.nrm.ave <- matrix(0, nrow = nrow(dds.cnt.nrm), ncol = ncol(dds.cnt.nrm)/3)
grps <- unique(sub("_.$", "", sampleName))
row.names(cnt.nrm.ave) <- row.names(dds.cnt.nrm)
colnames(cnt.nrm.ave)  <- grps
for(i in 1:length(grps)){
  grp <- grps[i]
  val <- apply(dds.cnt.nrm[,colnames(dds.cnt.nrm)==grp],1,mean)
  cnt.nrm.ave[,i] <- val
}

################# Part II: Naive WT vs KO

## all up

de.up.al <- unique(c(de.e10$up.lst, de.e14$up.lst, de.e16$up.lst, de.e19$up.lst))

cnt.nrm.up <- dds.cnt.nrm[row.names(dds.cnt.nrm) %in% de.up.al,]

ann_col <- data.frame(
  row.names = colnames(cnt.nrm.up),
  Group = groupName
)

ann_row_up <- data.frame(
  row.names = row.names(cnt.nrm.up),
  gd19.5 = ifelse(row.names(cnt.nrm.up) %in% de.e19$up.lst, "Yes", "No"),
  gd16.5 = ifelse(row.names(cnt.nrm.up) %in% de.e16$up.lst, "Yes", "No"),
  gd14.5 = ifelse(row.names(cnt.nrm.up) %in% de.e14$up.lst, "Yes", "No"),
  gd10.5 = ifelse(row.names(cnt.nrm.up) %in% de.e10$up.lst, "Yes", "No")
)

ann_color <- list(
  gd10.5 = c(Yes = "orange", No = "grey"),
  gd14.5 = c(Yes = "orange", No = "grey"),
  gd16.5 = c(Yes = "orange", No = "grey"),
  gd19.5 = c(Yes = "orange", No = "grey")
)

svg(file = "Fig.DEG_up_all_zscore.heatmap.svg", width = 9, height = 5)
pheatmap(
  cnt.nrm.up, 
  scale = "row", show_rownames = TRUE, cluster_cols = FALSE,
  labels_row = sub("^.*:","",row.names(cnt.nrm.up)),
  gaps_col = c(6,12,18),
  annotation_row = ann_row_up,
  annotation_legend = TRUE,
  annotation_colors = ann_color
)
dev.off()

boxplot(log10(cnt.nrm.up[ann_row_up$gd16.5=="Yes",]+1), las = 2)
boxplot(log10(cnt.nrm.up[ann_row_up$gd19.5=="Yes",]+1), las = 2)


## all down

de.dn.al <- unique(c(de.e10$dn.lst, de.e14$dn.lst, de.e16$dn.lst, de.e19$dn.lst))

cnt.nrm.dn <- dds.cnt.nrm[row.names(dds.cnt.nrm) %in% de.dn.al,]

ann_col <- data.frame(
  row.names = colnames(cnt.nrm.dn),
  Group = groupName
)
ann_row_dn <- data.frame(
  row.names = row.names(cnt.nrm.dn),
  gd19.5 = ifelse(row.names(cnt.nrm.dn) %in% de.e19$dn.lst, "Yes", "No"),
  gd16.5 = ifelse(row.names(cnt.nrm.dn) %in% de.e16$dn.lst, "Yes", "No"),
  gd14.5 = ifelse(row.names(cnt.nrm.dn) %in% de.e14$dn.lst, "Yes", "No"),
  gd10.5 = ifelse(row.names(cnt.nrm.dn) %in% de.e10$dn.lst, "Yes", "No")
)

ann_color <- list(
  gd10.5 = c(Yes = "orange", No = "grey"),
  gd14.5 = c(Yes = "orange", No = "grey"),
  gd16.5 = c(Yes = "orange", No = "grey"),
  gd19.5 = c(Yes = "orange", No = "grey")
)

svg(file = "Fig.DEG_dn_all_zscore.heatmap.svg", width = 10, height = 8)
pheatmap(
  cnt.nrm.dn, 
  scale = "row", show_rownames = TRUE, cluster_cols = FALSE,
  labels_row = sub("^.*:","",row.names(cnt.nrm.dn)),
  gaps_col = c(6,12,18),
  annotation_row = ann_row_dn, annotation_colors = ann_color
)
dev.off()

svg(file = "Fig.DEG_dn_all_zscore.heatmap.gd16only.svg", width = 5, height = 6)
pheatmap(
  cnt.nrm.dn[,13:18], 
  scale = "row", show_rownames = TRUE, cluster_cols = FALSE,
  labels_row = sub("^.*:","",row.names(cnt.nrm.dn))
)
dev.off()
